### PR TITLE
Cleanup/Re-work of README

### PR DESCRIPTION
**Pull request by hayden-cardwell**
My pull request to add SethForPrivacy's guide the other day got approved, so I decided to spend a bit of time cleaning up the Readme. Something to be said for Brevity, so here's the rough goals I had:

1. Make the steps easier to follow for someone with moderate technical experience, but not necessarily mining experience (think a r/monero member that's told they should be mining on P2Pool to fight a large pool having significant hash rate). 
2. Improve general readability/navigability (Numbered lists instead of bullet points, Table of Contents, code blocks for commands)
3. Create an overall more "polished" landing page for the project that inspires confidence in the software. 

Obviously this is a big change, so please let know if something doesn't sit right or if you're not a fan of a particular portion I re-worked.

### CODE REVIEW

**Comment by marvin8** [comment]
Since you are making these 'big' changes... from a complete stranger on the internet:

I'd like any information on why or when should I pick p2pool mini over standard p2pool mining... even just a link to some info would be great.

File: README.md [hunk 1]
```diff
@@ -1,19 +1,30 @@
 # Monero P2Pool
 
-Decentralized pool for Monero mining.
+Decentralized pool for Monero mining.
```
Comment (SChernykh) [inline README.md:3]: Unnecessary change?

**Comment by SChernykh** [comment]
@hayden-cardwell Looks good. Please you squash commits before I merge.

File: README.md (hunk 1 above)
Comment (hayden-cardwell) [reply README.md:3]: This is an unnecessary change, I wasn't aware of trailing spaces being carried into the changes.

**Comment by hayden-cardwell** [comment]
Thank you for the feedback! Shamefully I'm a git newbie and struggling to figure out the proper CLI options to have only one commit and only change the readme, but I'll revisit again tomorrow when I have more time to search through YouTube tutorials :)

**Comment by hayden-cardwell** [comment]
Also @marvin8, great feedback! This proposal was mainly to change formatting and improve readability, but I think having a small bit of info regarding P2Pool mini would be very helpful. If I get some time I'll try and track down some info and propose it as a separate request, assuming someone can proofread it for technical accuracy.

**Comment by SChernykh** [comment]
@hayden-cardwell you can use GitHub Desktop: clone your repository there, switch to History tab, select all commits (Ctrl + left mouse button) and then right click -> squash

**Comment by SChernykh** [comment]
Oh, now this PR has a merge commit which messed up things. I'll have to close it and submit the updated README myself (with you as an author).

**Comment by hayden-cardwell** [comment]
@SChernykh, my apologies! I'll do that moving forward. I had good intentions, but bad version control abilities.
